diff --git a/src/base/build_config.h b/src/base/build_config.h
index e5e017c222d..52c85a2a407 100644
--- a/src/base/build_config.h
+++ b/src/base/build_config.h
@@ -30,6 +30,5 @@
 // pthread_jit_write_protect is only available on arm64 Mac.
-#if defined(V8_HOST_ARCH_ARM64) && \
-    (defined(V8_OS_MACOS) || (defined(V8_OS_IOS) && TARGET_OS_SIMULATOR))
+#if defined(V8_HOST_ARCH_ARM64) && defined(V8_OS_MACOS)
 #define V8_HAS_PTHREAD_JIT_WRITE_PROTECT 1
 #else
 #define V8_HAS_PTHREAD_JIT_WRITE_PROTECT 0
